mimic/rest/rackconnect_v3_api.py

```diff
@@ -0,0 +1,447 @@
+from twisted.web.server import Request
+from zope.interface import implementer
+
+from mimic.catalog import Entry
+from mimic.catalog import Endpoint
+from mimic.imimic import IAPIMock
+from mimic.rest.mimicapp import MimicApp
+from mimic.util.helper import random_ipv4, seconds_to_timestamp
+
+
+Request.defaultContentType = 'application/json'
+timestamp_format = '%Y-%m-%dT%H:%M:%SZ'
+
+
+@implementer(IAPIMock, IPlugin)
+class RackConnectV3(object):
+    """
+    API mock object for RackConnect V3.
+
+    :ivar regions: The regions for which endpoints should be produced
+    :type regions: ``iterable``
+
+    :ivar int default_lbs: The number of default load balancers to be created
+        per tenant per region
+    """
+
+    def __init__(self, regions=("ORD",), default_lbs=1):
+        """
+        Construct a :class:`RackConnectV3` object.
+        """
+        self.regions = regions
+        self.default_lbs = default_lbs
+
+    def catalog_entries(self, tenant_id):
+        """
+        Catalog entry for RackConnect V3 endpoints.
+        """
+        return [
+            Entry(tenant_id, "rax:rackconnect", "rackconnect", [
+                Endpoint(tenant_id, region, text_type(uuid4()), prefix="v3")
+                for region in self.regions
+            ])
+        ]
+
+    def resource_for_region(self, region, uri_prefix, session_store):
+        """
+        Return an IResource implementing a public RackConnect V3 region
+        endpoint.
+        """
+        return RackConnectV3Region(
+            iapi=self,
+            uri_prefix=uri_prefix,
+            session_store=session_store,
+            region_name=region,
+            default_lbs=self.default_lbs).app.resource()
+
+
+@attributes(
+    [Attribute("id", default_factory=lambda: text_type(uuid4()),
+               instance_of=text_type),
+     Attribute("name", default_value=u"default", instance_of=text_type),
+     Attribute("port", default_value=80, instance_of=int),
+     Attribute("status", default_value=u"ACTIVE", instance_of=text_type),
+     Attribute("status_detail", default_value=None),
+     Attribute("virtual_ip", default_factory=random_ipv4),
+     Attribute('nodes', default_factory=list, instance_of=list)],
+    apply_with_cmp=False)
+class LoadBalancerPool(object):
+    """
+    Represents a RackConnecvt v3 Load Balancer Pool.
+
+    Load balancer pools cannot be created via the API, they have to just
+    already exist on the account.
+
+    :param text_type id: The ID of the load balancer pool - is randomly
+        generated if not provided
+    :param text_type name: The name of the load balancer pool - defaults to
+        'default'
+    :param int port: The incoming port of the load balancer - defaults to 80
+    :param text_Type status: The status of the load balancer - defaults to
+        "ACTIVE"
+    :param text_type status_detail: Any details about the status - defaults
+        to None
+    :param list nodes: :class:`LoadBalancerPoolNode`s
```

Comment:
virtual_ip is undocumented. Specifically I'm wondering what kind of IP this would realistically be, and if random_ipv4 needs to be specialized to produce the correct type.